docs/command_clu2adv.md

```diff
@@ -0,0 +1,196 @@
+
+## Usage
+
+If you want to convert a Terraform configuration from `mongodbatlas_cluster` to `mongodbatlas_advanced_cluster`, use the following command:
+```bash
+atlas terraform clusterToAdvancedCluster --file in.tf --output out.tf
+```
+
+you can also use shorter aliases, e.g.: 
+```bash
+atlas tf clu2adv -f in.tf -o out.tf
+```
+
+### Command Options
+
+- `--file` or `-f`: Input file path containing the `mongodbatlas_cluster` configuration
+- `--output` or `-o`: Output file path for the converted `mongodbatlas_advanced_cluster` configuration
+- `--replaceOutput` or `-r`: Overwrite the output file if it exists, or even use the same output file as the input file
+- `--watch` or `-w`: Keep the plugin running and watching for changes in the input file
+- `--includeMoved` or `-m`: Include the `moved blocks` in the output file
+
+## Examples
+
+You can find [here](https://github.com/mongodb-labs/atlas-cli-plugin-terraform/tree/main/internal/convert/testdata/clu2adv) some examples of input files (suffix .in.tf) and the corresponding output files (suffix .out.tf).
+
+## Dynamic Blocks
+
+`dynamic` blocks are used to generate multiple nested blocks based on a set of values. 
+Given the different ways of using dynamic blocks, we recommend reviewing the output and making sure it fits your needs.
+
+### Dynamic blocks in tags and labels
+
+You can use `dynamic` blocks for `tags` and `labels`. The plugin assumes that `for_each` has an expression which is evaluated to a `map` of strings.
+You can also combine the use of dynamic blocks in `tags` and `labels` with individual blocks in the same cluster definition, e.g.:
+```hcl
+tags {
+	key   = "environment"
+	value = var.environment
+}
+dynamic "tags" {
+	for_each = var.tags
+	content {
+		key   = tags.key
+		value = replace(tags.value, "/", "_")
+	}
+}
+```
+
+### Dynamic blocks in regions_config
+
+You can use `dynamic` blocks for `regions_config`. The plugin assumes that `for_each` has an expression which is evaluated to a `list` of objects.
+
+This is an example of how to use dynamic blocks in `regions_config`:
+```hcl
+replication_specs {
+  num_shards = var.replication_specs.num_shards
+  zone_name  = var.replication_specs.zone_name # only needed if you're using zones
+  dynamic "regions_config" {
+    for_each = var.replication_specs.regions_config
+    content {
+      priority        = regions_config.value.priority
+      region_name     = regions_config.value.region_name
+      electable_nodes = regions_config.value.electable_nodes
+      read_only_nodes = regions_config.value.read_only_nodes
+    }
+  }
+}
+```
+
+### Dynamic blocks in replication_specs
+
+You can use `dynamic` blocks for `replication_specs`. The plugin assumes that `for_each` has an expression which is evaluated to a `list` of objects.
```

Comment:
```suggestion
You can use `dynamic` blocks for `replication_specs`. The plugin assumes that the value of `for_each` is an expression which evaluates to a `list` of objects.
```
Broadly speaking, an 'expression' is not something which one 'has'. Furthermore, a key like 'for_each' does not typically 'have' anything other than its value, which may be an expression, etc.